src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -3159,7 +3159,7 @@ internal Buffer(IEnumerable<TElement> source, bool queryInterfaces = true)
 
         internal TElement[] ToArray()
         {
-            if (count == 0) return new TElement[0];
+            if (count == 0) return Array.Empty<TElement>();
```

Comment:
I think we considered changing this, but were not sure if it is enough benefit compared to potential for breaking code relying on distinct array instances. 
User might already have some allocations when hitting this path so one less empty array might not make a lot of improvement in typical code. The documentation says "An IEnumerable<T> to create an array from." If taken literally it seems to say that array is always created.
I wonder if there is a way to estimate if there is actual usage of this with expectation that array is created. Perhaps it can be done by scanning some publicly available code or something...

I think we should not change this just yet, but may worth adding a comment that it is being considered.